src/android/src/main/java/com/airbnb/android/react/lottie/LottieAnimationViewPropertyManager.java

```diff
@@ -0,0 +1,142 @@
+package com.airbnb.android.react.lottie;
+
+import android.util.Log;
+import android.widget.ImageView;
+
+import com.airbnb.lottie.LottieAnimationView;
+
+import org.json.JSONObject;
+
+import java.lang.ref.WeakReference;
+
+public class LottieAnimationViewPropertyManager {
+
+  private final WeakReference<LottieAnimationView> viewWeakReference;
+
+  private JSONObject animationJson;
+  private Float progress;
+  private Boolean loop;
+  private Float speed;
+
+  /**
+   * Should be set to true if one of the animationName related parameters has changed as result of latest
+   * reconciliation. We need to update the animation in this case.
+   */
+  private boolean animationNameDirty;
+
+  private String animationName;
+  private LottieAnimationView.CacheStrategy cacheStrategy;
+  private Boolean useHardwareAcceleration;
+  private ImageView.ScaleType scaleType;
+  private String imageAssetsFolder;
+  private Boolean enableMergePaths;
+
+  public LottieAnimationViewPropertyManager(LottieAnimationView view) {
+    this.viewWeakReference = new WeakReference<>(view);
+  }
+
+  public void setAnimationName(String animationName) {
+    this.animationName = animationName;
+    this.animationNameDirty = true;
+  }
+
+  public void setAnimationJson(String json) {
+    try {
+      this.animationJson = new JSONObject(json);
```

Comment:
Hold off on submitting this until #296 is merged so you can keep the animation as a string since 2.5.0 uses a JsonReader

Reply:
ok!